File: examples/aiogram/bot.py
"""aiogram example pointing at a local mtgo-bot-api server.

Run:  pip install aiogram
      BOT_TOKEN=<token> python examples/aiogram/bot.py
"""

import asyncio
import os
import sys

from aiogram import Bot, Dispatcher, F
from aiogram.client.default import DefaultBotProperties
from aiogram.types import Message
from aiogram.enums import ParseMode

# aiogram v3: set the base API URL via the session's base.
from aiogram.utils.token import (
    TokenValidationError,
)  # noqa: F401  (ensures import path exists)

API_ROOT = "http://localhost:8081"
token = os.environ.get("BOT_TOKEN")
if not token:
    sys.exit("BOT_TOKEN env var is required")

# aiogram v3 Bot accepts a `base` (aiohttp session) — the simplest override is
# building the Bot with a custom session whose base URL is the local server.
bot = Bot(
    token=token,
    default=DefaultBotProperties(parse_mode=ParseMode.MARKDOWN),
    session=None,  # set below
)

# Override the API base URL: aiogram builds "<base>/bot<token>/<method>".
bot.session.api_root = API_ROOT

dp = Dispatcher()


@dp.message(F.text == "/start")
async def cmd_start(message: Message) -> None:
    await message.answer("👋 Hello from mtgo-bot-api via *aiogram*!")


@dp.message(F.text == "/test")
async def cmd_test(message: Message) -> None:
    await message.answer("📦 aiogram round-trip OK")


@dp.message(F.text == "/inline")
async def cmd_inline(message: Message) -> None:
    from aiogram.utils.keyboard import InlineKeyboardBuilder

    kb = InlineKeyboardBuilder().button(text="Tap me", callback_data="tap").as_markup()
    await message.answer("Inline keyboard via aiogram:", reply_markup=kb)


@dp.callback_query(F.data == "tap")
async def on_tap(call) -> None:
    await call.answer("aiogram callback OK ✅")


async def main() -> None:
    me = await bot.get_me()
    print(
        f"✅ aiogram connected: @{me.username} ({me.id})  base={bot.session.api_root}"
    )
    print("📡 aiogram polling…")
    await dp.start_polling(bot)


if __name__ == "__main__":
    asyncio.run(main())

File: examples/grammy/bot.ts
/**
 * grammY example pointing at a local mtgo-bot-api server.
 * (A minimal version of examples/test-bot/ — see that for the full comparison harness.)
 *
 * Run: bun install grammy && BOT_TOKEN=<token> bun run examples/grammy/bot.ts
 */
import { Bot, InlineKeyboard } from "grammy";

const TOKEN = process.env.BOT_TOKEN ?? "";
const API_ROOT = process.env.API_ROOT ?? "http://localhost:8081";

if (!TOKEN) {
  console.error("BOT_TOKEN env var is required");
  process.exit(1);
}

const bot = new Bot(TOKEN, { client: { apiRoot: API_ROOT } });

bot.command("start", (ctx) =>
  ctx.reply("👋 Hello from mtgo-bot-api via *grammY*!", { parse_mode: "Markdown" })
);

bot.command("test", (ctx) => ctx.reply("📦 grammY round-trip OK"));

bot.command("inline", async (ctx) => {
  const kb = new InlineKeyboard().text("Tap me", "tap");
  await ctx.reply("Inline keyboard via grammY:", { reply_markup: kb });
});

bot.callbackQuery("tap", (ctx) => ctx.answerCallbackQuery({ text: "grammY callback OK ✅" }));

// Auto-reconnect so the bot survives server restarts.
async function runForever() {
  while (true) {
    try {
      const me = await bot.api.getMe();
      console.log(`✅ grammY connected: @${me.username} (${me.id})  base=${API_ROOT}`);
      console.log("📡 grammY polling…");
      await bot.start();
    } catch (err: any) {
      console.error(`⚠️ ${err?.message ?? err}. Reconnecting in 3s…`);
      await new Promise((r) => setTimeout(r, 3000));
    }
  }
}

runForever();
